src/ui/pages/Settings/Settings.tsx

```diff
@@ -48,18 +48,41 @@ import { Button } from 'src/ui/ui-kit/Button';
 import { Spacer } from 'src/ui/ui-kit/Spacer';
 import { useBackgroundKind } from 'src/ui/components/Background';
 import { openHrefInTabIfSidepanel } from 'src/ui/shared/openInTabIfInSidepanel';
+import { useWalletParams } from 'src/ui/shared/requests/useWalletParams';
+import { invariant } from 'src/shared/invariant';
 import { Security } from '../Security';
 import { BackupFlowSettingsSection } from './BackupFlowSettingsSection';
 import { PreferencesPage } from './Preferences';
 
+const ZERION_ORIGIN = 'https://app.zerion.io';
+
 function SettingsMain() {
   const { singleAddressNormalized } = useAddressParams();
   const navigate = useNavigate();
   const logout = useMutation({
     mutationFn: () => accountPublicRPCPort.request('logout'),
   });
 
+  const { data: currentWallet } = useQuery({
+    queryKey: ['wallet/uiGetCurrentWallet'],
+    queryFn: () => {
+      return walletPort.request('uiGetCurrentWallet');
+    },
+  });
+
+  const { mutate: acceptZerionOrigin } = useMutation({
+    mutationFn: async () => {
+      invariant(currentWallet, 'Current wallet not found');
```

Comment:
May be it is better to make a return here in case there is no wallet?
Or disable button or not render it while we are waiting for wallet. To exclude case when user clicks and see error
This is probably will not be an often case but anyway

Reply:
Let's not render the button if the is no `currentWallet`. In reality that should never happen anyway ✅